monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -381,37 +384,63 @@ def get_machine_aws_instance_id(hostname):
     def get_manual_monkey_hostnames():
         return [monkey["hostname"] for monkey in get_manual_monkeys()]
 
-    @staticmethod
-    def get_config_users():
-        return ConfigService.get_config_value(USER_LIST_PATH, True)
+    @classmethod
+    def get_config_users(cls):
+        usernames = []
+        configured_credentials = cls._credentials_repository.get_configured_credentials()
+        for credentials in configured_credentials:
+            usernames = chain(
+                usernames,
+                (
+                    identity
+                    for identity in credentials.identities
+                    if identity.credential_type == CredentialComponentType.USERNAME
+                ),
+            )
+        return [u.username for u in usernames]
 
-    @staticmethod
-    def get_config_passwords():
-        return ConfigService.get_config_value(PASSWORD_LIST_PATH, True)
+    @classmethod
+    def get_config_passwords(cls):
+        passwords = []
+        configured_credentials = cls._credentials_repository.get_configured_credentials()
+        for credentials in configured_credentials:
+            passwords = chain(
+                passwords,
+                (
+                    secret
+                    for secret in credentials.secrets
+                    if secret.credential_type == CredentialComponentType.PASSWORD
+                ),
+            )
 
-    @staticmethod
-    def get_config_exploits():
-        exploits_config_value = EXPLOITER_CLASSES_PATH
-        # TODO: Return default config here
-        default_exploits = ConfigService.get_default_config(False)
-        for namespace in exploits_config_value:
-            default_exploits = default_exploits[namespace]
-        exploits = ConfigService.get_config_value(exploits_config_value, True)
+        return [p.password for p in passwords]
 
-        if exploits == default_exploits:
-            return ["default"]
+    @classmethod
+    def get_config_exploits(cls):
+        agent_configuration = cls._agent_configuration_repository.get_configuration()
+        exploitation_configuration = agent_configuration.propagation.exploitation
+
+        enabled_exploiters = (
+            exploiter
+            for exploiter in chain(
+                exploitation_configuration.brute_force, exploitation_configuration.vulnerability
+            )
+        )
 
         return [
-            ExploiterDescriptorEnum.get_by_class_name(exploit).display_name for exploit in exploits
+            ExploiterDescriptorEnum.get_by_class_name(exploiter.name).display_name
+            for exploiter in enabled_exploiters
         ]
 
-    @staticmethod
-    def get_config_ips():
-        return ConfigService.get_config_value(SUBNET_SCAN_LIST_PATH, True)
+    @classmethod
+    def get_config_ips(cls):
```

Comment:
This is also redundant, as UI can request config and already has the schema and everything. But maybe let's keep it for now